File: quick_sort.rb
def quick_sort(array)
	if (array.length < 2) 
		return array
	else
		pivot = array.pop
		left = []
		right = []
		array.each do |a|
	 		a < pivot ? left.push(a) : right.push(a)
		end	
		return (quick_sort(left) + [pivot] + quick_sort(right))
	end
end
quick_sort([1,0,-1,3])

File: quickSort.js

//Do note that the O(log n) speed is a best-case/average time, in worst case scenarios it can be O(n2) depending on the implementation.

function quickSort(array) {
    if(array.length < 2) { return array}
    let pivot = array.shift()
    let left = []
    let right = []
    for(let i = 0; i < array.length; i++) {
        if(array[i] < pivot) {
            left.push(array[i])
        } else {
            right.push(array[i])
        }
    }
    return([...quickSort(left), pivot, ...quickSort(right)])
}

quickSort([1,0,-1,3])
